File: java/TuduManager/domain/src/test/java/net/seymourpoler/tudumanager/domain/todo/delete/DeleteTodoServiceShould.java
package net.seymourpoler.tudumanager.domain.todo.delete;

import net.seymourpoler.tudumanager.domain.ErrorCodes;
import org.assertj.core.api.AssertionsForInterfaceTypes;
import org.junit.Before;
import org.junit.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class DeleteTodoServiceShould {

    private IDeleteTodoService service;
    private IExistTodoRepository existTodoRepository;

    @Before
    public void setUp(){
        existTodoRepository = mock(IExistTodoRepository.class);
        service = new DeleteTodoService(existTodoRepository);
    }

    @Test
    public void
    return_error_when_is_not_found(){
        final Integer todoId = 3;
        when(existTodoRepository.exist(todoId)).thenReturn(false);

        var result = service.delete(todoId);

        AssertionsForInterfaceTypes
            .assertThat(result.errors().get(0).errorCode)
            .isEqualTo(ErrorCodes.NotFound);
    }
}

File: dotnet/react.mvc-core.postgres.dapper/TuduManager/test/TuduManayer.Domain.Test/Todo/Delete/DeleteTodoServiceShould.cs
using Moq;
using Shouldly;
using TuduManayer.Domain.Todo;
using TuduManayer.Domain.Todo.Delete;
using Xunit;

namespace TuduManayer.Domain.Test.Todo.Delete
{
    public class DeleteTodoServiceShould
    {
        private Mock<IExistTodoRepository> existRepository;
        private Mock<IDeleteTodoRepository> deleteRepository;
        private IDeleteTodoService service;

        public DeleteTodoServiceShould()
        {
            existRepository = new Mock<IExistTodoRepository>();
            deleteRepository = new Mock<IDeleteTodoRepository>();
            service = new DeleteTodoService(existRepository.Object, deleteRepository.Object);
        }

        [Fact]
        public void return_error_when_todo_is_not_found()
        {
            const int someTodoId = 3;
            existRepository
                .Setup(x => x.Exist(someTodoId))
                .Returns(false);

            var result = service.Delete(someTodoId);
            
            result.IsOk.ShouldBeFalse();
        }
        
        [Fact]
        public void delete_todo()
        {
            const int someTodoId = 3;
            existRepository
                .Setup(x => x.Exist(someTodoId))
                .Returns(true);

            var result = service.Delete(someTodoId);
            
            result.IsOk.ShouldBeTrue();
            deleteRepository.Verify(x => x.Delete(someTodoId));
        }
    }
}